src/proxy/src/main/java/ee/ria/xroad/proxy/ProxyMain.java

```diff
@@ -238,7 +240,23 @@ private static void createServices() throws Exception {
                 MessageLogProperties.isArchiveEncryptionEnabled(),
                 MessageLogProperties.isMessageLogEncryptionEnabled(),
                 ARCHIVE_GROUPING.name(),
-                getMessageLogArchiveEncryptionMembers());
+                getMessageLogArchiveEncryptionMembers(getMembers()));
+    }
+
+    private static List<ClientId> getMembers() throws Exception {
+        try {
+            return ServerConf.getMembers().stream()
+                    .filter(ProxyMain::isMemberRegistered)
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
+            log.error("Failed to get members from server configuration", e);
+            throw e;
+        }
+    }
+
+    private static boolean isMemberRegistered(ClientId member) {
```

Comment:
This filter can be removed. The list should include all local clients regardless of their status. There's no need to restrict the list to registered clients only.